network/agent/agent.ts

```diff
@@ -21,18 +22,36 @@ export function getAgent (uri: string, opts: AgentOptions) {
   return getNonProxyAgent(uri, opts)
 }
 
+function getClientCertificates(uri: string, opts: AgentOptions) {
+  const { host, hostOnlyDomain, pathname, nerf } = parseUri(uri)
+
+  if (host.endsWith(pathname)) {
+    return opts.clientCertificates?.[host];
+  }
+
+  const fullPath = `${nerf}${pathname !== '/' ? pathname : ''}`;
+  return opts.clientCertificates?.[fullPath] ?? opts.clientCertificates?.[nerf] ?? opts.clientCertificates?.[hostOnlyDomain];
```

Comment:
this is not how it works for auth token:

https://github.com/pnpm/pnpm/blob/4511aaf6e97c2128fe89889949fccf3a88a7146a/network/auth-header/src/index.ts#L39-L42